Modules/IO/NIFTI/test/itkNiftiImageIOTest13.cxx

```diff
@@ -0,0 +1,52 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+
+// Test to read small voxel NIFTI file which was triggering numerical instability
+// in IsAffine loading code
+int
+itkNiftiImageIOTest13(int argc, char * argv[])
+{
+  if (argc != 2)
+  {
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  try
+  {
+    using ReaderType = itk::ImageFileReader<ImageType>;
+    ReaderType::Pointer reader = ReaderType::New();
+    reader->SetFileName(argv[1]);
+    reader->Update();
+    ImageType::Pointer image = reader->GetOutput();
```

Comment:
Please consider using `itk::ReadImage` instead. Just one line of code should so it  😃

Reply:
Maybe some extra checks here to see that the read image is as expected...? Or at least, check that it is a valid image?

Reply:
https://github.com/InsightSoftwareConsortium/ITK/issues/2802

Reply:
> This test is to explicitly see if the NiftiIO isAffine check fails on the random image I had in my collection which was causing issues

I guess you could still add some checks that the read image is valid, for example: 

    if (image == nullptr)
    {
      std::cerr << "Error: The read image is null!\n";
      return EXIT_FAILURE;
    }

    if (image->GetBufferPointer() == nullptr)
    {
      std::cerr << "Error: the buffer of the read image is null!\n";
      return EXIT_FAILURE;
    }

    if (image->GetBufferedRegion().GetNumberOfPixels() == 0)
    {
      std::cerr << "Error: The read image has no pixels!\n";
      return EXIT_FAILURE;
    }

Right?

Reply:
@gdevenyi Did you still consider adding the extra checks that I suggested here? In order to ensure that the reading does actually yield a valid image?

```
if (image == nullptr)
{
  std::cerr << "Error: The read image is null!\n";
  return EXIT_FAILURE;
}

if (image->GetBufferPointer() == nullptr)
{
  std::cerr << "Error: the buffer of the read image is null!\n";
  return EXIT_FAILURE;
}

if (image->GetBufferedRegion().GetNumberOfPixels() == 0)
{
  std::cerr << "Error: The read image has no pixels!\n";
  return EXIT_FAILURE;
}

```

Reply:
This can be condiserably abbreviated via `itk::ReadImage` and `ITK_TRY_EXPECT_NO_EXCEPTION`. We can do that in a follow-up PR. Let's merge this, as it has dragged on for quite a long time.